types/express-serve-static-core/index.d.ts

```diff
@@ -440,7 +440,10 @@ interface Request extends http.IncomingMessage, Express.Request {
 
     method: string;
 
-    params: any;
+    params: {
+        [namedParameter: string]: string;
+        [captureGroup: number]: string;
+    };
```

Comment:
@timruffles @OliverJAsh 
`req.params` is a normal object, not an array but when the route is wildcard (`/*`) the keys are numbers so you can do `req.params[0]`:

![2017-07-19 12_41_27-server js test visual studio code](https://user-images.githubusercontent.com/10618781/28363214-ee039a58-6c7f-11e7-932c-ed40af1ab8e9.png)

So we can't enforce people to use `["0"]` bracket notation.